starter_code/components.js

```diff
@@ -0,0 +1,35 @@
+class Image extends React.Component{
+  constructor(){
+    super();
+    this.state = { image: "http://cdn2.business2community.com/wp-content/uploads/2016/01/QA-Danny-DeVito-008.jpg" }
+  }
+  _handleClick() {
+    if (this.state.image) {
```

Comment:
So I think there's a little confusion here over how to use the `!` operator. The point of using it (in this particular instance) is so we **don't** need to use a conditional at all. We're relying on the fact that our `this.state.showImage` variable is a boolean, meaning that when we put a `!` in front of it, it flips the value to to the opposite (i.e. `true` becomes `false`, `false` becomes `true`).

So instead of using both a conditional AND the `!` operator, we can do this:

```js
this.setState({ showImage: !this.state.showImage });
```

...and that's it. Now, if `showImage` *was* true, it gets set to false and vice versa. Obviously this won't work the same with your strategy where you're saving the whole URL string in the state since it's not a boolean.